src/FileList/FileExtInfoLoader.cc

```diff
@@ -425,120 +455,118 @@ void FileExtInfoLoader::FitsHeaderInfoToHeaderEntries(casacore::ImageFITSHeaderI
-                            }
-                        }
-
-                        auto header_entry = extended_info.add_header_entries();
-                        header_entry->set_name(name);
-                        *header_entry->mutable_value() = header_string;
-                        header_entry->set_entry_type(CARTA::EntryType::STRING);
-                        header_entry->set_comment(comment);
-                    }
-                    break;
-                }
-                case casacore::FITS::BIT:
-                case casacore::FITS::CHAR:
-                case casacore::FITS::COMPLEX:
-                case casacore::FITS::ICOMPLEX:
-                case casacore::FITS::DCOMPLEX:
-                case casacore::FITS::VADESC:
-                case casacore::FITS::NOVALUE:
-                default:
-                    break;
+                auto header_entry = extended_info.add_header_entries();
+                header_entry->set_name(name);
+                header_entry->set_comment(comment);
+                *header_entry->mutable_value() = header_string;
+                header_entry->set_entry_type(CARTA::EntryType::STRING);
+                break;
+            }
+            case casacore::FITS::COMPLEX: {
+                casacore::Complex value = fkw->asComplex();
+                std::string string_value = fmt::format("{} + {}i", value.real(), value.imag());
+
+                auto header_entry = extended_info.add_header_entries();
+                header_entry->set_name(name);
+                header_entry->set_comment(comment);
+                header_entry->set_entry_type(CARTA::EntryType::STRING);
+                break;
+            }
+            case casacore::FITS::ICOMPLEX: {
+                casacore::IComplex value = fkw->asIComplex();
+                std::string string_value = fmt::format("{} + {}i", value.real(), value.imag());
+
+                auto header_entry = extended_info.add_header_entries();
+                header_entry->set_name(name);
+                header_entry->set_comment(comment);
+                header_entry->set_entry_type(CARTA::EntryType::STRING);
+                break;
+            }
+            case casacore::FITS::DCOMPLEX: {
+                casacore::DComplex value = fkw->asDComplex();
+                std::string string_value = fmt::format("{} + {}i", value.real(), value.imag());
+
+                auto header_entry = extended_info.add_header_entries();
+                header_entry->set_name(name);
+                header_entry->set_comment(comment);
+                header_entry->set_entry_type(CARTA::EntryType::STRING);
+                break;
             }
+            case casacore::FITS::NOVALUE: {
+                auto header_entry = extended_info.add_header_entries();
+                header_entry->set_name(name);
+                header_entry->set_comment(comment);
+                header_entry->set_entry_type(CARTA::EntryType::STRING);
+                break;
```

Comment:
1. It looks like the calculated `string_value` not actually being used in any of these cases.
2. Would it be possible to refactor this case statement to move the common elements of the `header_entry` creation outside?